File: 2025-1/sesion-de-clase-20/SoftProg/sesion-de-clase-24/src/main/java/Identificador.java
/**
 *
 * @author eric
 */
public class Identificador {
    private static int id = 0;
    // private static final Object lock = new Object();

    public synchronized static void incrementar() {
//        synchronized (lock) {
//            id++;
//            System.out.println(id);
//        }
        id++;
        System.out.println(id);
    }

    public static int obtenerId() {
        return id;
    }
}

File: 2025-1/sesion-de-clase-24/net/Identificador.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sesion_de_clase_24_net
{
    class Identificador
    {
        private static int id = 0;
        private static readonly Object lockObj = new Object();

        public static void Incrementar() {
            lock (lockObj)
            {
                id++;
                Console.WriteLine(id);
            }

        }
        public static int ObtenerId() {
            lock (lockObj)
            {
                return id;
            }
        }
    }
}
